### PR TITLE
Add the ability to inject labels on the kustomization controller pods

### PR DESCRIPTION
**Pull request by mickael-carl**
Also remove an unused variable. This should make it possible to use Azure Workload Identity for Flux.

### CODE REVIEW

**Review by fcdm — APPROVED** [review]

**Comment by lassizci** [comment]
The template doesn't seem to use $pod_labels, so the variable seems ineffective.

**Comment by mickael-carl** [comment]
@lassizci thanks for catching that. Fixed now

File: kustomization.yaml.tpl [hunk 1]
```diff
@@ -15,6 +15,16 @@ patches:
     kind: ServiceAccount
     name: kustomize-controller
     labelSelector: app.kubernetes.io/part-of=flux
+- patch: |
+      apiVersion: apps/v1
+      kind: Deployment
+      spec:
+        template:
+          metadata:
+            labels: ${pod_labels}
```
Comment (lassizci) [inline kustomization.yaml.tpl:24]: one thing just crossed to my mind is that if we have `labels: []` would this actually then clear the default lables?

Comment (lassizci) [reply kustomization.yaml.tpl:24]: answering to myself: no. But at the same time patching doesn't seem to work without `metadata.name`.

Comment (mickael-carl) [reply kustomization.yaml.tpl:24]: The [docs](https://kubectl.docs.kubernetes.io/references/kustomize/glossary/#patchstrategicmerge) are unclear to me, but if I read them correctly I think you're right. I'll change that to just add those labels

Comment (lassizci) [reply kustomization.yaml.tpl:24]: I just dumped existing `kustomize-controller` and created `kustomization.yaml`. As that's a map, labels will be merged.

But we need to add `metadata.name`, so:
```
- patch: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: source-controller|image-reflector-controller
    . . .
```

and then for target `kind: Deployment` should be enough. Without `metadata.name` `kustomize build` would throw a bit cryptic error like:

```
$ kustomize build
Error: trouble configuring builtin PatchTransformer with config: `
patch: |-
  apiVersion: apps/v1
  kind: Deployment
  spec:
    template:
      metadata:
        labels: {}
target:
  kind: Deployment
`: unable to parse SM or JSON patch from [apiVersion: apps/v1
kind: Deployment
spec:
  template:
    metadata:
      labels: {}]
```

Comment (lassizci) [reply kustomization.yaml.tpl:24]: I did some ad-hoc testing with:
```
$ cat kustomization.yaml 
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
- test-deployment.yaml

patches:
  - patch: |-
      apiVersion: apps/v1
      kind: Deployment
      metadata:
        name: .*
      spec:
        template:
          metadata:
            labels:
              foo: bar
    target:
      kind: Deployment
```
and
```
$ cat test-deployment.yaml 
apiVersion: apps/v1
kind: Deployment
metadata:
  name: kustomize-controller
  namespace: flux-system
spec:
  template:
    metadata:
      annotations:
        prometheus.io/port: "8080"
        prometheus.io/scrape: "true"
      labels:
        app: kustomize-controller
    spec:
      containers: []
```

Comment (mickael-carl) [reply kustomization.yaml.tpl:24]: Thanks for all the testing, much appreciated 🙂 I'll make those changes.

Comment (mickael-carl) [reply kustomization.yaml.tpl]: Done!

**Pull request merged**